ramips: add support for YunCore AX820/HWAP-AX820

oe1rfc opened the pull request:
```
There are two versions which are identical apart from the enclosure:
  YunCore AX820: indoor ceiling mount AP with integrated antennas
  YunCore HWAP-AX820: outdoor enclosure with external (N) connectors

Hardware specs:
  SoC: MediaTek MT7621DAT
  Flash: 16 MiB SPI NOR
  RAM: 128MiB (DDR3, integrated)
  WiFi: MT7905DAN+MT7975DN 2.4/5GHz 2T2R 802.11ax
  Ethernet: 10/100/1000 Mbps x2 (WAN/PoE+LAN)
  LED: Status (green)
  Button: Reset
  Power: 802.11af/at PoE; DC 12V,1A
  Antennas: AX820(indoor): 4dBi internal; HWAP-AX820(outdoor): external

Flash instructions:
  The "OpenWRT support" version of the AX820 comes with a LEDE-based
  firmware with proprietary MTK drivers and a luci webinterface and
  ssh accessible under 192.168.1.1 on LAN; user root, no password.
  The sysupgrade.bin can be flashed using luci or sysupgrade via ssh,
  you will have to force the upgrade due to a different factory name.
  Remember: Do *not* preserve factory configuration!

MAC addresses as used by OEM firmware:
  use   address            source
  2g    44:D1:FA:*:0b      Factory 0x0004 (label)
  5g    46:D1:FA:*:0b      LAA of 2g
  lan   44:D1:FA:*:0c      Factory 0xe000
  wan   44:D1:FA:*:0d      Factory 0xe000 + 1
The wan MAC can also be found in 0xe006 but is not used by OEM dtb.

Due to different MAC handling in mt76 the LAA derived from lan is used
for 2g to prevent duplicate MACs when creating multiple interfaces.

Signed-off-by: Clemens Hopfer <openwrt@wireloss.net>
```

oe1rfc committed:
ramips: add support for YunCore AX820/HWAP-AX820

There are two versions which are identical apart from the enclosure:
  YunCore AX820: indoor ceiling mount AP with integrated antennas
  YunCore HWAP-AX820: outdoor enclosure with external (N) connectors

Hardware specs:
  SoC: MediaTek MT7621DAT
  Flash: 16 MiB SPI NOR
  RAM: 128MiB (DDR3, integrated)
  WiFi: MT7905DAN+MT7975DN 2.4/5GHz 2T2R 802.11ax
  Ethernet: 10/100/1000 Mbps x2 (WAN/PoE+LAN)
  LED: Status (green)
  Button: Reset
  Power: 802.11af/at PoE; DC 12V,1A
  Antennas: AX820(indoor): 4dBi internal; HWAP-AX820(outdoor): external

Flash instructions:
  The "OpenWRT support" version of the AX820 comes with a LEDE-based
  firmware with proprietary MTK drivers and a luci webinterface and
  ssh accessible under 192.168.1.1 on LAN; user root, no password.
  The sysupgrade.bin can be flashed using luci or sysupgrade via ssh,
  you will have to force the upgrade due to a different factory name.
  Remember: Do *not* preserve factory configuration!

MAC addresses as used by OEM firmware:
  use   address            source
  2g    44:D1:FA:*:0b      Factory 0x0004 (label)
  5g    46:D1:FA:*:0b      LAA of 2g
  lan   44:D1:FA:*:0c      Factory 0xe000
  wan   44:D1:FA:*:0d      Factory 0xe000 + 1
The wan MAC can also be found in 0xe006 but is not used by OEM dtb.

Due to different MAC handling in mt76 the LAA derived from lan is used
for 2g to prevent duplicate MACs when creating multiple interfaces.

Signed-off-by: Clemens Hopfer <openwrt@wireloss.net>

oe1rfc commented:
device bootlog:
```
>> Initializing DDR3

>> Applying AC timing parameters (try 1)
Expected DRAM size: 128MB

>> Setting DDR/CPU PLL ...
Using 3PLL mode with External loopback
DDR/CPU clock will be set to 1200MHz/880MHz

>> Starting DDR PLL calibration ...
DDR PLL2/3/4 feedback delay: 11/21/20
DDR 3PLL mode calibration passed

>> Starting DQS Gating Window Delay calibration
         0   8  16  24  32  40  48  56  64  72  80  88  96 104 112 120
      ----------------------------------------------------------------
0000:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
0001:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
0002:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
0003:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
0004:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
0005:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
0006:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
0007:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
0008:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
0009:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
000A:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
000B:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
000C:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
000D:|   0   0   0   0   0   0   0   0   0   0   0   1   1   1   1   1
000E:|   0   0   0   0   0   0   1   1   1   1   1   1   1   1   1   1
000F:|   0   1   1   1   1   1   1   1   1   1   0   0   0   0   0   0
0010:|   1   1   1   1   1   1   0   0   0   0   0   0   0   0   0   0
0011:|   1   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
0012:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
0013:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
0014:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
0015:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
0016:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
0017:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
0018:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
0019:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
001A:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
001B:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
001C:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
001D:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
001E:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
001F:|   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
Found coarse/fine delay value: 14/88

>> Starting DLE calibration
         0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15
      ----------------------------------------------------------------
0000:|   0   0   0   0   0   0   0   0   0   1   1   1   0   0   0   0
Optimal DQS factor: 10

>> Starting RX DQ per-bit delay calibration
Bit      0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15
      ----------------------------------------------------------------
Delay   13  10  13  13  13  11  14  10  10   8  11  10  10  11   9   9

>> Starting DQS delay calibration
Bit ( Range ) Center
--------------------
  0 (  1~ 58)     29
  1 (  1~ 57)     29
  2 (  1~ 58)     29
  3 (  1~ 57)     29
  4 (  2~ 60)     31
  5 (  1~ 57)     29
  6 (  1~ 58)     29
  7 (  1~ 57)     29
DQS0 input delay: 31

Bit ( Range ) Center
--------------------
  8 (  1~ 57)     29
  9 (  0~ 57)     28
 10 (  1~ 57)     29
 11 (  1~ 57)     29
 12 (  1~ 60)     30
 13 (  1~ 59)     30
 14 (  1~ 58)     29
 15 (  1~ 57)     29
DQS1 input delay: 30

>> Final DQ delay value
Bit      0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15
      ----------------------------------------------------------------
Delay   15  12  15  15  13  13  15  12  11  10  12  11  10  11  10  10

>> Starting TX DQ/QDS per-byte calibration
## Booting kernel from Legacy Image at bfc90000 ...
   Image Name:   MIPS OpenWrt Linux-5.4.132
   Image Type:   MIPS Linux Kernel Image (lzma compressed)
   Data Size:    2509862 Bytes = 2.4 MiB
   Load Address: 80001000
   Entry Point:  80001000
   Verifying Checksum ... OK
   Uncompressing Kernel Image ... OK
[    0.000000] Linux version 5.4.132 (datacop@devrandom) (gcc version 8.4.0 (OpenWrt GCC 8.4.0 r17182-831dc328fb)) #0 SMP Thu Jul 22 16:56:19 2021
[    0.000000] SoC Type: MediaTek MT7621 ver:1 eco:3
[    0.000000] printk: bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 0001992f (MIPS 1004Kc)
[    0.000000] MIPS: machine is YunCore AX820
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] VPE topology {2,2} total 4
[    0.000000] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.000000] MIPS secondary cache 256kB, 8-way, linesize 32 bytes.
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000]   HighMem  empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] percpu: Embedded 14 pages/cpu s26480 r8192 d22672 u57344
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 32480
[    0.000000] Kernel command line: console=ttyS0,115200 rootfstype=squashfs,jffs2
[    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes, linear)
[    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes, linear)
[    0.000000] Writing ErrCtl register=00041400
[    0.000000] Readback ErrCtl register=00041400
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 120528K/131072K available (6038K kernel code, 203K rwdata, 1276K rodata, 1236K init, 230K bss, 10544K reserved, 0K cma-reserved, 0K highmem)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] NR_IRQS: 256
[    0.000000] random: get_random_bytes called from start_kernel+0x36c/0x584 with crng_init=0
[    0.000000] CPU Clock: 880MHz
[    0.000000] clocksource: GIC: mask: 0xffffffffffffffff max_cycles: 0xcaf478abb4, max_idle_ns: 440795247997 ns
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 4343773742 ns
[    0.000008] sched_clock: 32 bits at 440MHz, resolution 2ns, wraps every 4880645118ns
[    0.007803] Calibrating delay loop... 586.13 BogoMIPS (lpj=2930688)
[    0.073975] pid_max: default: 32768 minimum: 301
[    0.078762] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.085962] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.096424] rcu: Hierarchical SRCU implementation.
[    0.101756] smp: Bringing up secondary CPUs ...
[    2.604094] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    2.604104] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    2.604117] MIPS secondary cache 256kB, 8-way, linesize 32 bytes.
[    2.604219] CPU1 revision is: 0001992f (MIPS 1004Kc)
[    0.166542] Synchronize counters for CPU 1: done.
[    2.307497] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    2.307505] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    2.307513] MIPS secondary cache 256kB, 8-way, linesize 32 bytes.
[    2.307569] CPU2 revision is: 0001992f (MIPS 1004Kc)
[    0.257274] Synchronize counters for CPU 2: done.
[    2.396890] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    2.396898] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    2.396906] MIPS secondary cache 256kB, 8-way, linesize 32 bytes.
[    2.396966] CPU3 revision is: 0001992f (MIPS 1004Kc)
[    0.342452] Synchronize counters for CPU 3: done.
[    0.372315] smp: Brought up 1 node, 4 CPUs
[    0.380582] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.390344] futex hash table entries: 1024 (order: 3, 32768 bytes, linear)
[    0.397333] pinctrl core: initialized pinctrl subsystem
[    0.404036] NET: Registered protocol family 16
[    0.435421] workqueue: max_active 576 requested for napi_workq is out of range, clamping between 1 and 512
[    0.446579] clocksource: Switched to clocksource GIC
[    0.452828] thermal_sys: Registered thermal governor 'step_wise'
[    0.453293] NET: Registered protocol family 2
[    0.463868] IP idents hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.471523] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 6144 bytes, linear)
[    0.479824] TCP established hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.487414] TCP bind hash table entries: 1024 (order: 1, 8192 bytes, linear)
[    0.494379] TCP: Hash tables configured (established 1024 bind 1024)
[    0.500828] UDP hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.507321] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.514452] NET: Registered protocol family 1
[    0.518777] PCI: CLS 0 bytes, default 32
[    0.756514] 4 CPUs re-calibrate udelay(lpj = 2924544)
[    0.763131] workingset: timestamp_bits=14 max_order=15 bucket_order=1
[    0.772687] random: fast init done
[    0.783584] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.789380] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.801765] mt7621_gpio 1e000600.gpio: registering 32 gpios
[    0.807607] mt7621_gpio 1e000600.gpio: registering 32 gpios
[    0.813338] mt7621_gpio 1e000600.gpio: registering 32 gpios
[    0.819576] Serial: 8250/16550 driver, 3 ports, IRQ sharing disabled
[    0.827177] printk: console [ttyS0] disabled
[    0.831429] 1e000c00.uartlite: ttyS0 at MMIO 0x1e000c00 (irq = 19, base_baud = 3125000) is a 16550A
[    0.840429] printk: console [ttyS0] enabled
[    0.840429] printk: console [ttyS0] enabled
[    0.848684] printk: bootconsole [early0] disabled
[    0.848684] printk: bootconsole [early0] disabled
[    0.860798] spi-mt7621 1e000b00.spi: sys_freq: 220000000
[    0.867280] spi-nor spi0.0: mx25l12805d (16384 Kbytes)
[    0.872456] 4 fixed-partitions partitions found on MTD device spi0.0
[    0.878836] Creating 4 MTD partitions on "spi0.0":
[    0.883615] 0x000000000000-0x000000030000 : "Bootloader"
[    0.890205] 0x000000030000-0x000000040000 : "Config"
[    0.896453] 0x000000050000-0x000000090000 : "Factory"
[    0.902893] 0x000000090000-0x000001000000 : "firmware"
[    0.909569] 2 uimage-fw partitions found on MTD device firmware
[    0.915473] Creating 2 MTD partitions on "firmware":
[    0.920486] 0x000000000000-0x000000264c66 : "kernel"
[    0.926939] 0x000000264c66-0x000000f70000 : "rootfs"
[    0.933088] mtd: device 5 (rootfs) set to be root filesystem
[    0.938904] 1 squashfs-split partitions found on MTD device rootfs
[    0.945066] 0x000000570000-0x000000f70000 : "rootfs_data"
[    0.952857] libphy: Fixed MDIO Bus: probed
[    0.996958] libphy: mdio: probed
[    1.000359] mt7530 mdio-bus:1f: MT7530 adapts as multi-chip module
[    1.008740] mtk_soc_eth 1e100000.ethernet: generated random MAC address 32:c4:d9:13:e5:26
[    1.017834] mtk_soc_eth 1e100000.ethernet eth0: mediatek frame engine at 0xbe100000, irq 21
[    1.028703] mt7621-pci 1e140000.pcie: Parsing DT failed
[    1.036377] NET: Registered protocol family 10
[    1.042286] Segment Routing with IPv6
[    1.046031] NET: Registered protocol family 17
[    1.050810] 8021q: 802.1Q VLAN Support v1.8
[    1.058465] mt7530 mdio-bus:1f: MT7530 adapts as multi-chip module
[    1.072198] libphy: dsa slave smi: probed
[    1.076636] mt7530 mdio-bus:1f wan (uninitialized): PHY [dsa-0.0:00] driver [Generic PHY]
[    1.086250] mt7530 mdio-bus:1f lan (uninitialized): PHY [dsa-0.0:01] driver [Generic PHY]
[    1.096465] mt7530 mdio-bus:1f: configuring for fixed/rgmii link mode
[    1.107713] DSA: tree 0 setup
[    1.110975] rt2880-pinmux pinctrl: pcie is already enabled
[    1.116631] mt7621-pci-phy 1e149000.pcie-phy: PHY for 0xbe149000 (dual port = 1)
[    1.124183] mt7621-pci-phy 1e14a000.pcie-phy: PHY for 0xbe14a000 (dual port = 0)
[    1.231585] mt7621-pci-phy 1e149000.pcie-phy: Xtal is 40MHz
[    1.237165] mt7621-pci-phy 1e14a000.pcie-phy: Xtal is 40MHz
[    1.342637] mt7621-pci 1e140000.pcie: pcie2 no card, disable it (RST & CLK)
[    1.349585] mt7621-pci 1e140000.pcie: PCIE0 enabled
[    1.354438] mt7621-pci 1e140000.pcie: PCIE1 enabled
[    1.359315] mt7621-pci 1e140000.pcie: PCI coherence region base: 0x60000000, mask/settings: 0xf0000002
[    1.368770] mt7621-pci 1e140000.pcie: PCI host bridge to bus 0000:00
[    1.375104] pci_bus 0000:00: root bus resource [io  0x1e160000-0x1e16ffff]
[    1.381973] pci_bus 0000:00: root bus resource [mem 0x60000000-0x6fffffff]
[    1.388843] pci_bus 0000:00: root bus resource [bus 00-ff]
[    1.394341] pci 0000:00:00.0: [0e8d:0801] type 01 class 0x060400
[    1.400371] pci 0000:00:00.0: reg 0x10: [mem 0x00000000-0x7fffffff]
[    1.406635] pci 0000:00:00.0: reg 0x14: [mem 0x60600000-0x6060ffff]
[    1.412943] pci 0000:00:00.0: supports D1
[    1.416952] pci 0000:00:00.0: PME# supported from D0 D1 D3hot
[    1.423034] pci 0000:00:01.0: [0e8d:0801] type 01 class 0x060400
[    1.429068] pci 0000:00:01.0: reg 0x10: [mem 0x00000000-0x7fffffff]
[    1.435311] pci 0000:00:01.0: reg 0x14: [mem 0x60610000-0x6061ffff]
[    1.441626] pci 0000:00:01.0: supports D1
[    1.445618] pci 0000:00:01.0: PME# supported from D0 D1 D3hot
[    1.452802] pci 0000:01:00.0: [14c3:7916] type 00 class 0x000280
[    1.458873] pci 0000:01:00.0: reg 0x10: [mem 0x00000000-0x000fffff 64bit pref]
[    1.466077] pci 0000:01:00.0: reg 0x18: [mem 0x00000000-0x00003fff 64bit pref]
[    1.473299] pci 0000:01:00.0: reg 0x20: [mem 0x00000000-0x00000fff 64bit pref]
[    1.480629] pci 0000:01:00.0: supports D1 D2
[    1.484880] pci 0000:01:00.0: PME# supported from D0 D1 D2 D3hot D3cold
[    1.491518] pci 0000:01:00.0: 2.000 Gb/s available PCIe bandwidth, limited by 2.5 GT/s x1 link at 0000:00:00.0 (capable of 4.000 Gb/s with 5 GT/s x1 link)
[    1.506601] pci 0000:00:00.0: PCI bridge to [bus 01-ff]
[    1.511812] pci 0000:00:00.0:   bridge window [io  0x0000-0x0fff]
[    1.517904] pci 0000:00:00.0:   bridge window [mem 0x60000000-0x600fffff]
[    1.524663] pci 0000:00:00.0:   bridge window [mem 0x60100000-0x602fffff pref]
[    1.531874] pci_bus 0000:01: busn_res: [bus 01-ff] end is updated to 01
[    1.538672] pci 0000:02:00.0: [14c3:7915] type 00 class 0x000280
[    1.544708] pci 0000:02:00.0: reg 0x10: [mem 0x00000000-0x000fffff 64bit pref]
[    1.551936] pci 0000:02:00.0: reg 0x18: [mem 0x00000000-0x00003fff 64bit pref]
[    1.559161] pci 0000:02:00.0: reg 0x20: [mem 0x00000000-0x00000fff 64bit pref]
[    1.566468] pci 0000:02:00.0: supports D1 D2
[    1.570736] pci 0000:02:00.0: PME# supported from D0 D1 D2 D3hot D3cold
[    1.577373] pci 0000:02:00.0: 2.000 Gb/s available PCIe bandwidth, limited by 2.5 GT/s x1 link at 0000:00:01.0 (capable of 4.000 Gb/s with 5 GT/s x1 link)
[    1.592440] pci 0000:00:01.0: PCI bridge to [bus 02-ff]
[    1.597679] pci 0000:00:01.0:   bridge window [io  0x0000-0x0fff]
[    1.603746] pci 0000:00:01.0:   bridge window [mem 0x60300000-0x603fffff]
[    1.610522] pci 0000:00:01.0:   bridge window [mem 0x60400000-0x605fffff pref]
[    1.617733] pci_bus 0000:02: busn_res: [bus 02-ff] end is updated to 02
[    1.624371] pci 0000:00:00.0: BAR 0: no space for [mem size 0x80000000]
[    1.630975] pci 0000:00:00.0: BAR 0: failed to assign [mem size 0x80000000]
[    1.637926] pci 0000:00:01.0: BAR 0: no space for [mem size 0x80000000]
[    1.644509] pci 0000:00:01.0: BAR 0: failed to assign [mem size 0x80000000]
[    1.651465] pci 0000:00:00.0: BAR 8: assigned [mem 0x60000000-0x600fffff]
[    1.658244] pci 0000:00:00.0: BAR 9: assigned [mem 0x60100000-0x602fffff pref]
[    1.665435] pci 0000:00:01.0: BAR 8: assigned [mem 0x60300000-0x603fffff]
[    1.672212] pci 0000:00:01.0: BAR 9: assigned [mem 0x60400000-0x605fffff pref]
[    1.679422] pci 0000:00:00.0: BAR 1: assigned [mem 0x60600000-0x6060ffff]
[    1.686184] pci 0000:00:01.0: BAR 1: assigned [mem 0x60610000-0x6061ffff]
[    1.692963] pci 0000:00:00.0: BAR 7: assigned [io  0x1e160000-0x1e160fff]
[    1.699742] pci 0000:00:01.0: BAR 7: assigned [io  0x1e161000-0x1e161fff]
[    1.706537] pci 0000:01:00.0: BAR 0: assigned [mem 0x60100000-0x601fffff 64bit pref]
[    1.714262] pci 0000:01:00.0: BAR 2: assigned [mem 0x60200000-0x60203fff 64bit pref]
[    1.721998] pci 0000:01:00.0: BAR 4: assigned [mem 0x60204000-0x60204fff 64bit pref]
[    1.729737] pci 0000:00:00.0: PCI bridge to [bus 01]
[    1.734680] pci 0000:00:00.0:   bridge window [io  0x1e160000-0x1e160fff]
[    1.741456] pci 0000:00:00.0:   bridge window [mem 0x60000000-0x600fffff]
[    1.748233] pci 0000:00:00.0:   bridge window [mem 0x60100000-0x602fffff pref]
[    1.755430] pci 0000:02:00.0: BAR 0: assigned [mem 0x60400000-0x604fffff 64bit pref]
[    1.763170] pci 0000:02:00.0: BAR 2: assigned [mem 0x60500000-0x60503fff 64bit pref]
[    1.770911] pci 0000:02:00.0: BAR 4: assigned [mem 0x60504000-0x60504fff 64bit pref]
[    1.778655] pci 0000:00:01.0: PCI bridge to [bus 02]
[    1.783596] pci 0000:00:01.0:   bridge window [io  0x1e161000-0x1e161fff]
[    1.790371] pci 0000:00:01.0:   bridge window [mem 0x60300000-0x603fffff]
[    1.797147] pci 0000:00:01.0:   bridge window [mem 0x60400000-0x605fffff pref]
[    1.804902] mt7530 mdio-bus:1f: Link is Up - 1Gbps/Full - flow control off
[    1.805280] hctosys: unable to open rtc device (rtc0)
[    1.824744] VFS: Mounted root (squashfs filesystem) readonly on device 31:5.
[    1.835921] Freeing unused kernel memory: 1236K
[    1.840482] This architecture does not have kernel memory protection.
[    1.846905] Run /sbin/init as init process
[    2.615689] init: Console is alive
[    2.619399] init: - watchdog -
[    3.652274] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    3.799571] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    3.816931] init: - preinit -
[    4.775030] mtk_soc_eth 1e100000.ethernet eth0: configuring for fixed/rgmii link mode
[    4.783423] mtk_soc_eth 1e100000.ethernet eth0: Link is Up - 1Gbps/Full - flow control rx/tx
[    4.791910] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[    4.967677] random: jshn: uninitialized urandom read (4 bytes read)
[    5.032318] random: jshn: uninitialized urandom read (4 bytes read)
[    5.066163] random: jshn: uninitialized urandom read (4 bytes read)
[    5.270909] mt7530 mdio-bus:1f lan: configuring for phy/gmii link mode
[    5.277928] 8021q: adding VLAN 0 to HW filter on device lan
Press the [f] key and hit [enter] to enter failsafe mode
Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
[    9.449404] mount_root: jffs2 not ready yet, using temporary tmpfs overlay
[    9.460188] urandom-seed: Seed file not found (/etc/urandom.seed)
[    9.594208] procd: - early -
[    9.597219] procd: - watchdog -
[   10.176716] procd: - watchdog -
[   10.180503] procd: - ubus -
[   10.240354] procd: - init -
Please press Enter to activate this console.
[   10.701670] kmodloader: loading kernel modules from /etc/modules.d/*
[   10.787654] Loading modules backported from Linux version v5.10.42-0-g65859eca4dff
[   10.795206] Backport generated by backports.git v5.10.42-1-0-gbee5c545
[   10.823357] xt_time: kernel timezone is -0000
[   10.876341] mt7621-pci 1e140000.pcie: bus=1 slot=0 irq=23
[   10.881785] pci 0000:00:00.0: enabling device (0006 -> 0007)
[   10.887453] mt7915e_hif 0000:01:00.0: enabling device (0000 -> 0002)
[   10.891410] urngd: v1.0.2 started.
[   10.897666] mt7621-pci 1e140000.pcie: bus=2 slot=1 irq=24
[   10.903091] pci 0000:00:01.0: enabling device (0006 -> 0007)
[   10.908766] mt7915e 0000:02:00.0: enabling device (0000 -> 0002)
[   11.033414] random: crng init done
[   11.036865] random: 7 urandom warning(s) missed due to ratelimiting
[   11.112318] mt7915e 0000:02:00.0: HW/SW Version: 0x8a108a10, Build Time: 20201105222230a
[   11.112318] 
[   11.637924] mt7915e 0000:02:00.0: WM Firmware Version: ____000000, Build Time: 20201105222304
[   11.675182] mt7915e 0000:02:00.0: WA Firmware Version: DEV_000000, Build Time: 20201105222323
[   11.798575] mt7915e 0000:02:00.0: Invalid MAC address, using random address b2:b6:84:3d:ea:e0
[   11.826336] PPP generic driver version 2.4.2
[   11.832175] NET: Registered protocol family 24
[   11.847478] kmodloader: done loading kernel modules from /etc/modules.d/*
[   30.685415] jffs2_scan_eraseblock(): End of filesystem marker found at 0x0
[   30.692568] jffs2_build_filesystem(): unlocking the mtd device... 
[   30.692575] done.
[   30.700875] jffs2_build_filesystem(): erasing all blocks after the end marker... 
[   32.455845] mtk_soc_eth 1e100000.ethernet eth0: Link is Down
[   32.484632] mtk_soc_eth 1e100000.ethernet eth0: configuring for fixed/rgmii link mode
[   32.492990] mtk_soc_eth 1e100000.ethernet eth0: Link is Up - 1Gbps/Full - flow control rx/tx
[   32.501977] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[   32.509807] mt7530 mdio-bus:1f lan: configuring for phy/gmii link mode
[   32.516957] 8021q: adding VLAN 0 to HW filter on device lan
[   32.529685] br-lan: port 1(lan) entered blocking state
[   32.534873] br-lan: port 1(lan) entered disabled state
[   32.541446] device lan entered promiscuous mode
[   32.546076] device eth0 entered promiscuous mode
[   32.592223] mt7530 mdio-bus:1f wan: configuring for phy/gmii link mode
[   32.599680] 8021q: adding VLAN 0 to HW filter on device wan
[   36.757114] mt7530 mdio-bus:1f wan: Link is Up - 1Gbps/Full - flow control off
[   36.764372] IPv6: ADDRCONF(NETDEV_CHANGE): wan: link becomes ready
[   87.096311] done.
[   87.098293] jffs2: notice: (1655) jffs2_build_xattr_subsystem: complete building xattr subsystem, 0 of xdatum (0 unchecked, 0 orphan) and 0 of xref (0 dead, 0 orphan) found.
[   87.235467] overlayfs: upper fs does not support tmpfile.
```

Once this PR is merged I would open a PR for backporting it to 21.02.

mrkiko commented:
Hi!!

Is there any recovery method available in this device? And maybe usable without UART?

Thanks!

oe1rfc commented:
Hi,
although the uboot has an integrated web-based "MediaTek U-Boot System Recovery Mode" which accepts sysupgrade images, I could only trigger it by forcing a failed tftp boot via uart, which might also start on a bad kernel image.
Holding or toggling the reset button while power on does not do anything at all.

There is a YunCore-style 4-pin 3.3V UART which is well documented on the silkscreen on the PCB.

Ah, and I also forgot, there is an outdoor-version [HWAP-AX820](https://www.yuncore.com.cn/en/pro/swap/181.html) with external N connectors and surge arrestors, which is otherwise completely identical, I'll update the commit description.

adschm commented on target/linux/ramips/dts/mt7621_yuncore_ax820.dts:
```diff
@@ -0,0 +1,134 @@
+		led-failsafe = &led_status_green;
+		led-running = &led_status_green;
+		led-upgrade = &led_status_green;
+		label-mac-device = &wifi0;
```
This typically does not work without directly assigning the MAC address in DTS.

Use 02_network for label_mac instead.

oe1rfc replied:
thanks, `label-mac-device` and `wifi0` label removed and label detection moved to `02_network`.

adschm commented on target/linux/ramips/dts/mt7621_yuncore_ax820.dts:
```diff
@@ -0,0 +1,134 @@
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
```
This can very probably go faster.

oe1rfc replied:
ACK, updated to `80000000`, which seems to run fine in my tests.

adschm commented on target/linux/ramips/dts/mt7621_yuncore_ax820.dts:
```diff
@@ -0,0 +1,134 @@
+				read-only;
+			};
+
+			factory: partition@50000 {
```
0x40000 to 0x50000 is missing

oe1rfc replied:
you're right, I didn't notice this. It's actually not used in the factory firmware and empty in the devices I have, I added a note in the dts.

adschm commented on target/linux/ramips/dts/mt7621_yuncore_ax820.dts:
```diff
@@ -0,0 +1,134 @@
+			};
+
+			partition@90000 {
+			compatible = "denx,uimage";
```
broken indent.

oe1rfc replied:
thanks, fixed.

adschm commented on target/linux/ramips/dts/mt7621_yuncore_ax820.dts:
```diff
@@ -0,0 +1,134 @@
+};
+
+&pcie1 {
+	wifi0: wifi@0,0 {
```
Why "wifi0" and not "wifi1"? But since label-mac-device does not work, this DT label can probably be dropped anyway.

oe1rfc replied:
The mt7915 exposes two PHYs from a single pcie device. I've dropped the label.

adschm commented on target/linux/ramips/dts/mt7621_yuncore_ax820.dts:
```diff
@@ -0,0 +1,134 @@
+			label = "wan";
+			nvmem-cells = <&macaddr_factory_e000>;
+			nvmem-cell-names = "mac-address";
+			mac-address-increment = <1>;
```
No useful address in 0xe006?

oe1rfc replied:
The correct wan mac address is in `0xe006`, but for some reason the factory DTB uses `0xe000` with `mac-address-increment 1`. I wasn't sure if `0xe006` will always be correct, so I've implemented it just like the factory firmware does.

I can change this, if reading it from `0xe006` would be better.

adschm replied:
No, keep the incremented version. If factory does not use 0xe006, we cannot rely on it.

adschm replied:
But consider adding that very fact after the MAC address table in the commit message.

adschm commented on target/linux/ramips/dts/mt7621_yuncore_ax820.dts:
```diff
@@ -0,0 +1,134 @@
+	#size-cells = <1>;
+
+	macaddr_factory_e000: macaddr@e000 {
+		reg = <0x3fff4 0x6>;
```
Hmm, that's not 0xe000 ...

oe1rfc replied:
Ugh you're right, thanks m( fixed.

adschm commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -83,6 +83,17 @@ define Build/ubnt-erx-factory-image
+			"serial": [".*"], \
+			"supported_devices":["mt7621-rfb-ax-nor"] \
+		}' | fwtool -I - $@
+endef
```
What's supposed to do? Hacking in metadata is a NAK from me.

oe1rfc replied:
ACK, I wasn't sure if this makes sense, removed and updated flash instructions.

The factory firmware detects the device as `mt7621-rfb-ax-nor` but uses the sysupgrade metadata format otherwise, with this the factory.bin could be directly flashed.
There are quite a few devices/vendors that do this, is there any proper way to build factory-specific sysupgrade images?

adschm replied:
Have you tried adding
SUPPORTED_DEVICES += mt7621-rfb-ax-nor
?

oe1rfc replied:
I'm leaving this as-is, I don't think its useful to have a factory firmware for this device, since the other YunCore devices don't have one either.

adschm commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -1616,6 +1627,17 @@ define Device/youku_yk-l2
 
+define Device/yuncore_ax820
+  $(Device/dsa-migration)
+  IMAGE_SIZE := 7872k
```
That's a boldly wrong copy-paste either for the partitions or the IMAGE_SIZE.

oe1rfc replied:
You're right, somehow I even mis-calculated the wrong image size when checking, fixed.

adschm commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -1616,6 +1627,17 @@ define Device/youku_yk-l2
+  DEVICE_MODEL := AX820
+  DEVICE_PACKAGES := kmod-mt7915e
+  IMAGES += factory.bin
+  IMAGE/factory.bin := append-kernel | append-rootfs | pad-rootfs | yuncore-ax820-metadata | check-size
```
Max. 74 chars per line.

oe1rfc replied:
removed due to removed yuncore-ax820-metadata hack.

adschm commented on target/linux/ramips/mt7621/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac:
```diff
@@ -25,7 +25,8 @@ case "$board" in
-	jcg,q20)
+	jcg,q20|\
+	yuncore,ax820)
 		[ "$PHYNBR" = "1" ] && \
```
You should comment in your commit message on how MAC addresses for WiFi are set (as it seems you have 2g/5g with one radio here?)

oe1rfc replied:
Thanks, I've updated the commit message. Even though the 2.4GHz/phy0 MAC is the label MAC address, 5GHz is not even set by the factory firmware and randomly generated.

nbd168 replied:
Only setting the local bit on the second PHY's MAC address and changing nothing else causes MAC address conflicts when adding multiple VIFs per PHY.
This can be fixed by also incrementing the last byte like this:
https://pastebin.com/CgyLStnf

oe1rfc replied:
Thanks, took me a while to realize that this won't collide with `lan` because of the local bit.
I've updated the PR in the same code style that's used a few lines above for `dlink,dir-853-r1`: https://github.com/openwrt/openwrt/blob/f12413744f821e8219414e2ef3cff93ead263f41/target/linux/ramips/mt7621/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac#L28-L33

adschm replied:
Well, if vendor firmware uses a random address, we should do so as well. Using the increment will result in having MAC address collisions between different devices in certain cases. (Devices with subsequent MAC addresses do happen in real life.)

oe1rfc replied:
@adschm the resulting MAC (before setting the local bit) is already used by `lan` on the same device, so there should be no overlaps with other devices.
The phy1 MAC of the factory firmware did not seem to be derived from the label mac, but I'm not 100% sure if it changes at every reboot. I might be able to check this in a few days. but imho BSSIDs should not change at each reboot.

oe1rfc replied:
(for reference) full explanation here: https://github.com/openwrt/openwrt/pull/4390#issuecomment-1102944208

adschm commented:
>   Antennas: AX820(indoor): 4dBi internal; HWAP-AX820(outdoor): external

This information about the two "variants" is a little hidden. Consider adding one sentence right at the beginning (before specs) in the commit message where you just state that these two devices are identical except for the antenna configuration ...

oe1rfc commented:
@adschm many thanks for your detailed review, I've updated the PR with your requested changes, but i'm not a 100% sure about the MAC address for `lan`.

oe1rfc commented:
Also, I just realized that `dsa-migration` isn't needed in the device definition, since this is a new device anyway, removed.

adschm commented:
> Also, I just realized that `dsa-migration` isn't needed in the device definition, since this is a new device anyway, removed.

dsa-migration IS needed since board.d/05_compat-version sets version 1.1 unconditionally.

oe1rfc commented:
> dsa-migration IS needed since board.d/05_compat-version sets version 1.1 unconditionally.

Ok, thanks, I read this wrong on a different PR, fixed (and rebased).

adschm commented on target/linux/ramips/dts/mt7621_yuncore_ax820.dts:
```diff
@@ -0,0 +1,137 @@
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
```
&state_default is missing?

oe1rfc replied:
thanks, added below, please review if this seems correct.

adschm commented on target/linux/ramips/dts/mt7621_yuncore_ax820.dts:
```diff
@@ -0,0 +1,137 @@
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <80000000>;
```
fast-read missing?

oe1rfc replied:
thanks, added.

adschm commented on target/linux/ramips/mt7621/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac:
```diff
@@ -25,9 +25,12 @@ case "$board" in
+	jcg,q20|\
+	yuncore,ax820)
+		if [ "$PHYNBR" = "1" ]; then
+			base_mac=$(macaddr_add "$(mtd_get_mac_binary Factory 0x4)" 1)
```
Take 0xe000 directly instead of incrementing 0x4 by 1.

Please also add a comment to the commit message about this (after the other MAC address stuff):

`To avoid a random MAC address for 5 GHz, we deliberately chose to set it based on LAN address instead.`

adschm replied:
BTW you are changing q20 here as well. This should not happen. Just add a separate block.

adschm commented:
Should be good after the remarks are addressed.

Neustradamus commented:
@oe1rfc: Have you seen @adschm remarks?

danyanya commented:
Hello everybody, what's the current stage of this PR? 
@oe1rfc could you please make an update?

T-X commented:
Hi @oe1rfc, is setting the LAN/WAN mac via the DTS working for you? I think I have a very similar development board here, at least it has the same SoC and wifi chips and flash layout, where I get random MAC addresses on LAN/WAN. Unless I'm adding something like this to 02_network:

```
+       <board>)
+               label_mac=$(mtd_get_mac_binary Factory 0x4)
+               lan_mac=$(mtd_get_mac_binary Factory 0xe000)
+               wan_mac=$(mtd_get_mac_binary Factory 0xe006)
+               ;;
```

oe1rfc commented:
@T-X yes, reading the macs only in the DTS with `mtd-mac-address/mtd-mac-address-increment` is definitely working for me.

sorry about the silence, I'm still interested in merging this PR, I'll rebase it and write a comment on how the mtk MAC handling works different from mt76 in the next 1-2 weeks.

aleksander0m commented:
I've been using this patch for some time now on top of 21.02, and I'm seeing quite some issues with wrongly flashed sysupgrade files, which end up in mostly unusable systems reporting errors like these ones in the log:
```
Tue Apr 19 10:40:53 2022 kern.err kernel: [  528.947968] SQUASHFS error: Unable to read fragment cache entry [3970c2]
Tue Apr 19 10:40:53 2022 kern.err kernel: [  528.954675] SQUASHFS error: Unable to read page, block 3970c2, size d514
Tue Apr 19 10:40:53 2022 kern.err kernel: [  528.961475] SQUASHFS error: Unable to read fragment cache entry [3970c2]
Tue Apr 19 10:40:53 2022 kern.err kernel: [  528.968183] SQUASHFS error: Unable to read page, block 3970c2, size d514
Tue Apr 19 10:40:53 2022 kern.err kernel: [  528.974963] SQUASHFS error: Unable to read fragment cache entry [3970c2]
Tue Apr 19 10:40:53 2022 kern.err kernel: [  528.981669] SQUASHFS error: Unable to read page, block 3970c2, size d514
```
Has anyone experienced this type of errors as well?

oe1rfc commented:
@aleksander0m I just checked the dmesg of 18 devices I'm running with this patch backported to 21.02 but with `spi-max-frequency = <10000000>;` instead of `80000000`. @adschm remarked that `m25p,fast-read;` was in fact missing, I've just fixed this, I guess that this is the root cause for the error. I haven't got an AX820 here currently, but I'll test this this week.

@adschm thanks for your review, I've fixed your other requested changes, please have a look at this, I would really like to get this PR finally merged.

I have also adapted the reading of the PHY1/5g MAC by directly reading the LAA `lan` mac from `Factory 0xe000` instead of adding 1 to the PHY0/2g mac. This is needed as Felix found because deriving the 5g MAC from the 2G mac results in collisions if you have multiple interfaces, eg:
```
phy0/2g:
wlan0   44:d1:fa:b2:36:0b
wlan0-1 46:d1:fa:b2:36:0b
wlan0-2 42:d1:fa:b2:36:0b
phy1/5g:
wlan1   46:d1:fa:b2:36:0b  # 2g (wlan0) LAA; collision with wlan0-1
wlan1-1 42:d1:fa:b2:36:0b  # collision with wlan0-2
wlan1-2 4e:d1:fa:b2:36:0b  # collision with wlan0-3
```
the proprietary MTK drivers enumerate the interface MACs differently, that's why they`re not suffering this problem:
```
2g:
ra0 44:D1:FA:B2:36:0B
ra1 46:D1:FA:92:36:0B
ra2 46:D1:FA:A2:36:0B
5g:
rax0 46:D1:FA:B2:36:0B  # 2g (ra0) LAA
rax1 46:D1:FA:C2:36:0B
rax2 46:D1:FA:D2:36:0B
```
I think this problem also exists with other devices that use `macaddr_setbit_la "$(mtd_get_mac_binary Factory 0x4)"` to set the phy1 mac from phy0, that's out of the scope of this PR. If dresired i can address this in an additional PR/patchset later.

f00b4r0 commented:
Tested, working for me.
And it nicely avoids MAC collision with multiple wlans

aleksander0m commented:
> @aleksander0m I just checked the dmesg of 18 devices I'm running with this patch backported to 21.02 but with spi-max-frequency = <10000000>; instead of 80000000. @adschm remarked that m25p,fast-read; was in fact missing, I've just fixed this, I guess that this is the root cause for the error. I haven't got an AX820 here currently, but I'll test this this week.

Thanks! will try that

aleksander0m commented:
> > @aleksander0m I just checked the dmesg of 18 devices I'm running with this patch backported to 21.02 but with spi-max-frequency = <10000000>; instead of 80000000. @adschm remarked that m25p,fast-read; was in fact missing, I've just fixed this, I guess that this is the root cause for the error. I haven't got an AX820 here currently, but I'll test this this week.
> 
> Thanks! will try that

Adding `m25p,fast-read;` and leaving `spi-max-frequency = <80000000>;` seems to solve the issue; I've done several rebuilds and all seem to go well for now. Thanks!

oe1rfc commented:
thanks for verification!

hauke commented:
Thank you for your patch, I applied it to master in 4891b865380e2b7f32acf0893df9c1ca9db8d4ea.

oe1rfc commented:
@hauke thank you!

Djfe commented:
According to YunCore's website, the HWAP-AX820 uses 8MB of flash instead of 16mb and 128mb of ram instead of 256mb, so be careful if you try this image unaltered on the outdoor variant. it's everything but the same hardware.
the cpu is an MT7621DAT instead MT7621AT

[Outdoor](https://www.yuncorenet.com/collection/hwap-ax820-wi-fi6)
[Indoor](https://www.yuncorenet.com/item/ax820-wi-fi6/11ax-1800mbps-wi-fi-standard-mtk-chipset-high-power-industrial-ceiling-wireless-access-point)

oe1rfc commented:
@Djfe that's interesting, in all devices I have from YunCore they use the same PCBA and also the datasheets I got from them are identical for indoor/outdoor. If that is the case there might even be a 8MB indoor variant, as they might shuffle boards.
